### PR TITLE
Remote file handling

### PR DESCRIPTION
**Pull request by cerasamson**

### CODE REVIEW

File: src/Extension/KernelExtensions.cs [hunk 1]
```diff
@@ -32,18 +34,62 @@ public static class KernelExtensions
 
         public static T UseProgressiveLearning<T>(this T kernel) where T : Kernel, IKernelCommandHandler<ParseNotebook>
         {
-            var argument = new Argument<FileInfo>("file");
-            
+            Option<string> fromUrlOption = new Option<string>(
+                "--from-url",
```
Comment (colombod) [inline src/Extension/KernelExtensions.cs:38]: to be safer this should be a Uri object instead of string

**Review by colombod — CHANGES_REQUESTED** [review]

File: src/Extension/KernelExtensions.cs [hunk 2]
```diff
@@ -32,18 +34,62 @@ public static class KernelExtensions
...
+                "--from-file",
+                description: "Specify lesson source file",
+                parseArgument: result =>
+                {
+                    var filePath = result.Tokens.Single().Value;
+                    var fromUrlResult = result.FindResultFor(fromUrlOption);
+
+                    if (fromUrlResult is not null)
+                    {
+                        result.ErrorMessage = $"The {fromUrlResult.Token.Value} and {(result.Parent as OptionResult).Token.Value} cannot be used together";
```
Comment (AngleQian) [inline src/Extension/KernelExtensions.cs:51]: should be the ... and ... options cannot be used together.
Might be good to add a test for this

File: src/Extension/KernelExtensions.cs [hunk 3]
```diff
@@ -32,18 +34,62 @@ public static class KernelExtensions
 
         public static T UseProgressiveLearning<T>(this T kernel) where T : Kernel, IKernelCommandHandler<ParseNotebook>
         {
-            var argument = new Argument<FileInfo>("file");
-            
+            Option<Uri> fromUrlOption = new Option<Uri>(
+                "--from-url",
```
Comment (AngleQian) [inline src/Extension/KernelExtensions.cs:38]: is there any good way to test --from-url?

Comment (jonsequitur) [reply src/Extension/KernelExtensions.cs:38]: What is the behavior you want to test?

Comment (AngleQian) [reply src/Extension/KernelExtensions.cs:38]: does #!start-lesson --from-url work?

File: src/Extension/KernelExtensions.cs [hunk 4]
```diff
@@ -32,18 +34,62 @@ public static class KernelExtensions
...
+
             var startCommand = new Command("#!start-lesson")
             {
-                argument
+                fromFileOption,
+                fromUrlOption
             };
 
-            startCommand.Handler = CommandHandler.Create<FileInfo, KernelInvocationContext>(async (file, context) =>
+            startCommand.Handler = CommandHandler.Create<string, FileInfo, KernelInvocationContext>(async (fromUrl, fromFile, context) =>
```
Comment (AngleQian) [inline src/Extension/KernelExtensions.cs:73]: shouldn't this be type Uri?

File: src/Extension/KernelExtensions.cs (hunk 3 above)
Comment (jonsequitur) [reply src/Extension/KernelExtensions.cs:38]: Are you looking to test the parsing behavior, the ability to retrieve the data from a URL, other details?

I think we have good examples in the tests for the `KeyValueStoreKernel`.

**Review by lytesfazion — APPROVED** [review]

**Pull request merged**